tests/components/test_kernel_config_component.py

```diff
@@ -15,7 +19,6 @@ def test_config_dict():
     return {
         "section1": {
             "list": [1, 2, 3],
-            # "tuple": (1, "b"), # yaml cannot deal with tuple
```

Comment:
But did it use to then? Is that something we should keep? 
For Model ConfigComponent, I don't know a lot of models that use yaml but for the hydromt build/update config, this means that none of the @hydromt_step functions can have tuple in their arguments (also valid for plugins).

Reply:
I don't know, this comment was added in a PR before I joined (#292 ) What I could do is cast tuples to lists, when writing to file as these are functionally identical in python, although this would require a lot more work as then I'd have to traverse the whole dictionary. Trying to write tuples to yaml produces a valid but weird syntax and trying to read tuples leads it to just becoming a string: 
```python
In [10]: yaml.safe_load(r"""a:
    ...:     (1,2,3)""")
Out[10]: {'a': '(1,2,3)'}

In [11]: yaml.dump({'a':(1,2,3)})
Out[11]: 'a: !!python/tuple\n- 1\n- 2\n- 3\n'
```

Users can still set tuples in their python code through the `set` function, but given that it's not a feature of yaml I don't think this should be our concern.

Reply:
If this was already not supported before then we can ignore :) Just wanted to make we were not removing existing functionality with the transfer to components